Xext/panoramiX.c

```diff
@@ -1064,28 +1065,30 @@ ProcXineramaQueryScreens(ClientPtr client)
         swapl(&rep.length);
         swapl(&rep.number);
     }
-    WriteToClient(client, sizeof(xXineramaQueryScreensReply), &rep);
+
+    struct x_rpcbuf rpcbuf = { .swapped = client->swapped };
 
     if (!noPanoramiXExtension) {
-        xXineramaScreenInfo scratch;
         int i;
-
         FOR_NSCREENS_BACKWARD(i) {
-            scratch.x_org = screenInfo.screens[i]->x;
-            scratch.y_org = screenInfo.screens[i]->y;
-            scratch.width = screenInfo.screens[i]->width;
-            scratch.height = screenInfo.screens[i]->height;
-
-            if (client->swapped) {
+            xXineramaScreenInfo scratch = {
+                .x_org = screenInfo.screens[i]->x,
+                .y_org = screenInfo.screens[i]->y,
+                .width = screenInfo.screens[i]->width,
+                .height = screenInfo.screens[i]->height,
+            };
+            if (rpcbuf.swapped) {
                 swaps(&scratch.x_org);
                 swaps(&scratch.y_org);
                 swaps(&scratch.width);
                 swaps(&scratch.height);
             }
-            WriteToClient(client, sz_XineramaScreenInfo, &scratch);
+            x_rpcbuf_write_CARD8s(&rpcbuf, (CARD8*)&scratch, sizeof(scratch));
```

Comment:
Missing error check. Also why write CARD8s instead of using `x_rpcbuf_write_binary_pad` when the fields aren't CARD8s. This feels like a hack.

Reply:
you're right. fixed.